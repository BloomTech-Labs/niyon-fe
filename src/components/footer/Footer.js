import React from 'react'
import { Link } from 'react-router-dom'
import { makeStyles } from '@material-ui/core/styles'
import BottomNavigation from '@material-ui/core/BottomNavigation'
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction'
import HomeIcon from '@material-ui/icons/Home'
import SearchIcon from '@material-ui/icons/Search'
import PersonIcon from '@material-ui/icons/Person'
import SettingsIcon from '@material-ui/icons/Settings'
import Menu from '@material-ui/core/Menu'
import { signOut } from '../apiStuff/signout'
import { darkMode } from '../apiStuff/darkMode'
import MenuItem from '@material-ui/core/MenuItem'
import Brightness4Icon from '@material-ui/icons/Brightness4'
import ExitToAppIcon from '@material-ui/icons/ExitToApp'

const useStyles = makeStyles({
  root: {
    width: '100%',
    position: 'fixed',
    bottom: 0,
    color: 'primary'
  }
})

function Footer (props) {
  const classes = useStyles()
  const [anchorEl, setAnchorEl] = React.useState(null)
  const handleClick = (event) => {
    setAnchorEl(event.currentTarget)
  }
  const handleClose = () => {
    setAnchorEl(null)
  }

  return (
    <div>
      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem onClick={(e) => darkMode(e)}>
          <Brightness4Icon />
          <div>&nbsp;Dark Mode</div>
        </MenuItem>
        <MenuItem onClick={(e) => signOut(e)}>
          <ExitToAppIcon />
          <div>&nbsp;Log Out</div>
        </MenuItem>
      </Menu>
      <BottomNavigation value={props.value} showLabels className={classes.root}>
        <BottomNavigationAction
          className="icon"
          component={Link}
          to="/home"
          label="Home"
          icon={<HomeIcon />}
        />
        <BottomNavigationAction
          component={Link}
          to="/profile"
          label="Profile"
          icon={<PersonIcon />}
        />
        <BottomNavigationAction
          component={Link}
          to="/search"
          label="Search"
          icon={<SearchIcon />}
        />
        <BottomNavigationAction
          onClick={handleClick}
          label="Settings"
          icon={<SettingsIcon />}
        />
      </BottomNavigation>
    </div>
  )
}

export default Footer
