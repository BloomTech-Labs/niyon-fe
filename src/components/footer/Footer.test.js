import React from 'react';
import toJSON from 'enzyme-to-json';
import { createShallow } from '@material-ui/core/test-utils';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import Footer from './Footer';

const setUp = (props={}) => {
  let shallowWrapper = createShallow();
  const wrapper = shallowWrapper(<Footer {...props} />);
  return wrapper;
}


describe('<Footer /> component testing', () => {
   let component;
   beforeEach(() => {
       component = setUp();
   })

   it('should match the snapshot test for  <Footer /> component', () => {
         expect(toJSON(component)).toMatchSnapshot();
   })
  it('should render <Footer /> component correctly', () => {
         expect (component.exists()).toBe(true);
   });  

});