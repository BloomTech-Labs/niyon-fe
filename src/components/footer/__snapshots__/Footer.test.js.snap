// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> component testing should match the snapshot test for  <Footer /> component 1`] = `
<div>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    id="simple-menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <Brightness4Icon />
      <div>
         Dark Mode
      </div>
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      onClick={[Function]}
    >
      <ExitToAppIcon />
      <div>
         Log Out
      </div>
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
  <WithStyles(ForwardRef(BottomNavigation))
    className="makeStyles-root-1"
    showLabels={true}
  >
    <WithStyles(ForwardRef(BottomNavigationAction))
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      icon={<UNDEFINED />}
      label="Home"
      to="/home"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      icon={<UNDEFINED />}
      label="Profile"
      to="/profile"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      icon={<UNDEFINED />}
      label="Search"
      to="/search"
    />
    <WithStyles(ForwardRef(BottomNavigationAction))
      icon={<UNDEFINED />}
      label="Settings"
      onClick={[Function]}
    />
  </WithStyles(ForwardRef(BottomNavigation))>
</div>
`;
